parameterspace/configspace_utils.py

```diff
@@ -0,0 +1,132 @@
+"""Initialize a `ParameterSpace` from a `ConfigSpace` JSON dictionary."""
+
+from typing import List, Optional
+
+import parameterspace as ps
+from parameterspace.condition import Condition
+from parameterspace.utils import verify_lambda
+
+
+def _escape_parameter_name(name: str) -> str:
+    """Replace colons with underscores.
+
+    Colons are incompatible as ParameterSpace parameter names.
+    """
+    return name.replace(":", "_")
+
+
+def _get_condition(conditions: List[dict], parameter_name: str) -> Optional[Condition]:
+    """Construct a lambda function that can be used as a ParameterSpace condition from a
+    ConfigSpace conditions list given a specific target parameter name.
+    """
+    condition = Condition()
+
+    varnames = []
+    function_texts = []
+    for cond in conditions:
```

Comment:
Not sure how quickly this becomes relevant, but there are more conditions in ConfigSpace, e.g. `GreaterThanCondition` and `LessThanCondition`. Additionally, there are also `And/OrConjunction` to combine conditions. In practice, I don't think these are super relevant as the "IN" and "NOTIN"(?) already cover most cases.

Reply:
Following https://automl.github.io/ConfigSpace/main/api/conditions.html there are

- ✅ EqualsCondition
- ❌ NotEqualsCondition
- ❌ LessThanCondition
- ❌ GreaterThanCondition
- ✅ InCondition
- ❌ AndConjunction
- ❌ OrConjunction

I'll integrate the missing ones since support should be trivial.

Regarding the conjunctions, I'm less sure how trivial this is because with that one can have arbitrary hierarchies of conditions, right? So I might opt to raise a `NotImplementedError` in the presence of conjunctions and test for that for now.

Reply:
The Conjunction will be non-trivial for sure. As you mentioned, they do support a hierarchy of conditions and more conjunctions, which requires some thought and is probably not used in most cases. At least for now, I don't think we really need the Conjunctions.

Reply:
I created #51 for additional transparency and us not to forget about it.